src/Service.Tests/Unittests/SqlMetadataProviderUnitTests.cs

```diff
@@ -97,20 +97,24 @@ public async Task CheckExceptionForBadConnectionStringForMsSql(string connection
         }
 
         /// <summary>
-        /// <code>Do: </code> Tests with different combinations of connection string to ensure
-        /// tableprefix generated is correctly.
-        /// <code>Check: </code> Making sure table prefix matches expected prefix.
+        /// <code>Do: </code> Tests with different combinations of data base, schema, and table names
+        /// to validate that the correct full table name with prefix is generated. For example if
+        /// databaseName = TestDB, schemaName = model, and tableName = TrainedModel, then correct would
+        /// mean [TestDB].[model].[TrainedModel], and any other form would be incorrect.
+        /// <code>Check: </code> Making sure table name with prefix matches expected name with prefix.
         /// </summary>
         [DataTestMethod]
-        [DataRow("", "", "")]
-        [DataRow("", "model", "[model]")]
-        [DataRow("TestDB", "", "[TestDB]")]
-        [DataRow("TestDB", "model", "[TestDB].[model]")]
-        public void CheckTablePrefix(string databaseName, string schemaName, string expectedPrefix)
+        [DataRow("", "", "", "[]")]
```

Comment:
is a table name allowed to be an empty string? in the `SqlMetadataProvider.GetTableNameWithPrefix(...)` function, there is a "IsNullOrEmpty" check for dbname and schema, but not for table name. Also, should we not expect GetTableNameWithPrefix to return a prefix, even with an empty table name?

Reply:
Table names are not allowed to be empty in sql as far as I know, but this is just a unit test that is checking if the formatting is done correctly. I did not create these cases originally, but modified them to fit the bug fix. I am apprehensive to remove these cases, even though I don't think theyre valid for actual use in SQL.


I can change the name if you don't like the idea of a prefix being empty.